sdk/spec_file_generator.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+type protocol string
+
+const (
+	protoTCP       protocol = "tcp"
+	protoNamedPipe protocol = "npipe"
+)
+
+func writeSpec(name, address string, proto protocol) (string, error) {
+
+	var pluginSpecDir string
+	if runtime.GOOS == "windows" {
+		pluginSpecDir = ([]string{filepath.Join(os.Getenv("programdata"), "docker", "plugins")})[0]
```

Comment:
I'm not sure if I understand correctly. I copied over code responsible for creating the spec directory from the file you linked, but I'm not sure if that fixes the problem?